lib/email/receiver.rb

```diff
@@ -230,6 +243,23 @@ def log_and_validate_user(user)
       raise SilencedUserError if user.silenced?
     end
 
+    def get_all_recipients(mail)
+      recipients = []
+
+      %i(to cc bcc).each do |field|
+        next if mail[field].blank?
+
+        mail[field].each do |address_field|
+          begin
+            address_field.decoded
+            recipients << address_field.address.downcase
+          end
+        end
+      end
+
+      recipients
```

Comment:
Is it possible `recipients` might contain duplicate email addresses? Should we do a `.uniq`?

Reply:
This is a very interesting question. No email client allowed me to duplicate the email address, but I assume it can be possible.